MapboxCoreNavigation/EventDetails.swift

```diff
@@ -352,7 +352,9 @@ enum EventDetailsError: Error {
 
 extension EventDetails {
     func asDictionary() throws -> [String: Any] {
-        let data = try JSONEncoder().encode(self)
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
```

Comment:
I think I was mistaken in https://github.com/mapbox/mapbox-navigation-ios/issues/2618#issue-699592630: so far, we’ve only identified that the values of this one key are in the wrong case, but the keys aren’t necessarily in the wrong case. After all, the general feedback event code hasn’t changed for quite a long time. Looking at the backend’s event schema, I think camel case is actually correct for the keys.

Reply:
Let's remove this then until it's clear it's needed.